src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -64,7 +64,15 @@ main() {
     filtered_inputs=""
     
     @@RUN_INPUTS@@
-    nextflow -trace nextflow.plugin $nf_advanced_opts -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name run-${NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
+
+    # parse dnanexus-job.json to get job output destination
+    # and use that dxpath as the nextflow params.outdir
+    # TODO: override publishDir in user's main.nf to `publishDir=params.outdir`
```

Comment:
To make this work it requires the publishDir in user's workflow to be `params.outdir`. If not the file would be stored in the local path inside the container but never get uploaded to the job destination given by `dx run`

We might need to change the plugin code to recognize this param. Alternatively, we can embed this param in a config_file and force use this config with `-C config_file`. But if we do that we need to merge it with user's config, otherwise the user's config will be ignored.

Reply:
we can use --publishDir=params.outdir, we dont need config file to override a value.

Though, if user had their own publishDir in their config, this would override the config value

Reply:
https://www.nextflow.io/docs/latest/cli.html?highlight=directive#pipeline-parameters
I am not sure that `--publishDir` would work cuz double-dashed will be referred as params.publishDir. Could you share an example that we could override the directives like publishDir from command line? @mhrvol

Reply:
@YuxinShi0423 If we merged your code, the example that I posted in the JIRA ticket APPS-1357 works in the same way as when running locally?

Reply:
I misunderstand what publishDir referred to back then.

I wonder, is it possible to see whether publishDir was used from the plugin?
Would creating job outputs solve our problem? (are files deleted if set as output of a job?)

Reply:
- is it possible to see whether publishDir was used from the plugin?
yes but it need to be evaluated if defined with some runtime params (see the case 4 above). I don't have enough time to fully investigate how to fetch and handle them now.
- Would creating job outputs solve our problem?
when subjobs are created the output specs are not specified in the api call `job/new` (and cannot be, see https://documentation.dnanexus.com/developer/api/running-analyses/job-input-and-output#output and
https://documentation.dnanexus.com/developer/api/running-analyses/applets-and-entry-points#api-method-job-new
could you elaborate on "creating job outputs"?
@mhrvol